### PR TITLE
Refactor/xmit record dao

### LINKED ISSUE #158: DcpMonitor application not surviving database connection exceed time.
**Describe the bug**

It was reported that the DcpMonitor doesn't survive connection timeouts.

**To Reproduce**
1. download/install/configured database for DCPMonitor tables
2. run application and wait
3. exceed time error will show in logs, monitor no longer stores xmit records.

**Expected behavior**
Application interacts with the pool correctly.

** ServerSide (please complete the following information):**
 - OS: any (CWMS Database)
 - Version: 7.0.4

**Additional context**
Already fixed in a local branch and running tests, XmitRecordDao wasn't used within a try-with-resource block so the connection wasn't getting close regularly.

### PR DESCRIPTION
**Pull request by MikeNeilson**
## Problem Description

May fix #158 . 
Migrate XmitRecordDAO to proper bind variables and connection usage.

Describe the problem you are trying to solve.

## Solution

- No longer hold on to PreparedStatements; while this does improve performance on a single connection it doesn't work in a connection pool.
- Use bind variables for every operation
- Create an "inTransaction" handler that allows using our DAO object function on a single Connection in a defined transaction. (Not as critical here but there's a few other areas that will benefit given we generally do delete-reinsert for a lot of updates.)
- Created a new integration-test Configuration function to be able to specify a specific DAO is required to be supported.

## how you tested the change

Create a simple integration test. I'll be expanding the test to a more reasonable volume of data but this initial single message
got the setup working.

## Where the following done:

- [ ] Tests. Check all that apply:
   - [ ] Unit tests created or modified that run during ant test.
   - [x] Integration tests created or modified that run during integration testing
         (Formerly called regression tests.)
   - [ ] Test procedure descriptions for manual testing
- [ ] Was relevant documentation updated?
- [ ] Were relevant config element (e.g. XML data) updated as appropriate

If you aren't sure leave unchecked and we will help guide you to want needs changing where.

### CODE REVIEW

File: src/main/java/opendcs/dao/XmitRecordDAO.java
[the remark this answers is not in the document]
Comment (MikeNeilson) [reply src/main/java/opendcs/dao/XmitRecordDAO.java:1]: No, the performance impact of that particular caching would be minimal anyways. Though Dave would probably disagree as all the database vendors promote doing that; it just doesn't work in the context of a connection pool at all as you have to cache a prepared statement per connection; which can just up and go away.

File: src/main/java/opendcs/dao/XmitRecordDAO.java [hunk 1]
```diff
@@ -79,1039 +49,1070 @@
...
+            }
+            catch (SQLException ex)
+            {
+                logger.warning("Unable to large message chunk " + xr.toString()
+                             + " because " + ex.getLocalizedMessage());
+            }
+
+        }
+    }
+
```
[the remark this answers is not in the document]
Comment (MikeNeilson) [reply src/main/java/opendcs/dao/XmitRecordDAO.java:432]: I'll add a specific test for this before merging in, but that looks reasonable.

[the remark this answers is not in the document]
Comment (MikeNeilson) [reply src/main/java/opendcs/dao/XmitRecordDAO.java:432]: Oh the first 4000 bytes are stored in the primary table, only the remaining bytes are stored in the secondary table; hence why it looks like they went missing.

[the remark this answers is not in the document]
Comment (MikeNeilson) [reply src/main/java/opendcs/dao/XmitRecordDAO.java:432]: There's an issue, #167, to just migrate the table to use clob/text directly; given this logic isn't used very often I'm not worried about fixing it now.

**Review by ktarbet — APPROVED** [review]

**Pull request merged**